### ISSUE TITLE
When will the .NET 3.1 runtimes be removed from the buildpack?

### ISSUE THREAD

**Issue by miguellira** [cloudfoundry/dotnet-core-buildpack#622]
This is **_not_** an issue. I'm simply looking for timelines and/or guidance for upgrading applications currently targeting .NET Core 3.1.

[According to Microsoft](https://dotnet.microsoft.com/en-us/platform/support/policy/dotnet-core#lifecycle), .NET Core 3.1 will lose support on **December 12, 2022**.

Does this mean the subsequent buildpack release after this date will _remove_ the .NET 3.1 SDK and runtime(s)?

Recently, in [release v2.3.42](https://github.com/cloudfoundry/dotnet-core-buildpack/releases/tag/v2.3.42) .NET 5 (_which was never an LTS_) was **removed** - although the release notes only mentioned them as "deprecated".

Bottom line, should developers using your buildpack be completely off of .NET 3.1 before the end of its LTS schedule or will the SDK and runtimes remain for some time after this date?

**Comment by ryanmoran** [comment]
> Bottom line, should developers using your buildpack be completely off of .NET 3.1 before the end of its LTS schedule or will the SDK and runtimes remain for some time after this date?

You should expect that at any point after the EOS date for .NET Core 3.1, the buildpack will no longer include .NET 3.1. We likely won't remove it immediately, but you can expect that it will likely be removed with the next release after that date.

> Recently, in [release v2.3.42](https://github.com/cloudfoundry/dotnet-core-buildpack/releases/tag/v2.3.42) .NET 5 (which was never an LTS) was removed - although the release notes only mentioned them as "deprecated".

We're sorry for the confusion caused by the release notes related to the removal of .NET 5. The intention, if not the actuality, was to say that the buildpack would no longer contain those versions.
